test/client_test.rb

```diff
@@ -129,27 +129,24 @@ class ClientTest < TinyTds::TestCase
       end
     end
 
-    it 'must run this test to prove we account for dropped connections' do
-      skip
+    it 'raises TinyTds exception with sql batch timeout due to network failure' do
+      skip if ENV['CI'] && ENV['APPVEYOR_BUILD_FOLDER'] # only CI using docker
       begin
-        client = new_connection :login_timeout => 2, :timeout => 2
+        client = new_connection timeout: 2
         assert_client_works(client)
-        STDOUT.puts "Disconnect network!"
-        sleep 10
-        STDOUT.puts "This should not get stuck past 6 seconds!"
+        docker_container('pause')
         action = lambda { client.execute('SELECT 1 as [one]').each }
         assert_raise_tinytds_error(action) do |e|
           assert_equal 20003, e.db_error_number
           assert_equal 6, e.severity
           assert_match %r{timed out}i, e.message, 'ignore if non-english test run'
         end
       ensure
-        STDOUT.puts "Reconnect network!"
-        sleep 10
```

Comment:
Keeping at least a little sleep might be a good idea, I don't know if we can rely on SQL Server coming online instantly when the container is unpaused. I actually put in a sleep after each docker command just to be sure. What about an optional keyword parameter to the `docker_container` method?

Reply:
Good point. Are you suggesting this `docker_container('sleep', wait_for: 1)`?

Reply:
Yep